editor.ts

```diff
@@ -270,12 +283,24 @@ namespace microcode {
                 icon: icons.get("reset"),
                 style: ButtonStyles.BorderedPurple,
                 ariaId: "reset",
-                x: Screen.RIGHT_EDGE - 12,
+                x: 0,
                 y: 8,
                 onClick: () => this.saveAndCompileProgram(),
             })
             this.progdef = this.app.load(SAVESLOT_AUTO)
             if (!this.progdef) this.progdef = new ProgramDefn()
+
+            this.layout()
+        }
+
+        private layout() {
+            this.scrollroot.xfrm.localPos = new Vec2(
+                Screen.LEFT_EDGE,
+                Screen.TOP_EDGE + TOOLBAR_HEIGHT + TOOLBAR_MARGIN
+            )
+            this.scrollDest.copyFrom(this.scrollroot.xfrm.localPos)
+            this.pageBtn.xfrm.localPos.x = Screen.RIGHT_EDGE - 32
```

Comment:
@pelikhan I was going to suggest this for relayout of the toolbar. Is it not working? You might try resetting hudroot's localPos, if Screen.TOP_EDGE changed. `this.hudroot.xfrm.localPos = new Vec2(0, Screen.TOP_EDGE)`

Reply:
i went for extreme and spanning editors on each page/screen size